test/realm_test.dart

```diff
@@ -1394,6 +1396,166 @@ Future<void> main([List<String>? args]) async {
     expect(await realmIsCancelled, isTrue);
     expect(progressReturned, isFalse);
   });
+
+  void addDataForCompact(Realm realm) {
+    realm.write(() {
+      for (var i = 0; i < 2500; i++) {
+        realm.add(Task(ObjectId()));
+      }
+    });
+
+    realm.write(() => realm.deleteAll<Task>());
+
+    realm.write(() {
+      for (var i = 10; i < 20; i++) {
+        realm.add(Task(ObjectId()));
+      }
+    });
+  }
+
+  Future<int> createRealmForCompact(Configuration config) async {
+    var realm = getRealm(config);
+    
+    if (config is FlexibleSyncConfiguration) {
+      realm.subscriptions.update((mutableSubscriptions) {
+        mutableSubscriptions.add(realm.all<Task>());
+      });
+      await realm.subscriptions.waitForSynchronization();
```

Comment:
We should avoid doing that as this introduces unpleasant dependencies between tests. I.e. if we have other tests that are adding `Task` objects, those will get synchronized here and we'll start the test with non-0 number of tasks. The way to avoid this is for example adding a `differentiator` field which works similar to a partition. So you do something like:

```dart
final differentiator = ObjectId();
realm.subscriptions.update((mutableSubs) {
  mutableSubs.add(realm.all<Task>().filter('differentiator = $0', differentiator);
});

addDataFroCompact(realm, differentiator);
```

That way, you're certain that every time you execute the test, it'll do the same thing and have the exact same number of objects.

Reply:
I agree. For this reason I created the Product class. So that I can add the differentiator in product name.

Reply:
will fix